src/app/cluster/details/cluster/mla/alertmanager-config/alertmanager-config-dialog/component.ts

```diff
@@ -21,13 +21,13 @@ import {AlertmanagerConfig} from '@shared/entity/mla';
 import {getIconClassForButton} from '@shared/utils/common';
 import _ from 'lodash';
 import {encode, decode} from 'js-base64';
-import {Subject} from 'rxjs';
+import {Observable, Subject} from 'rxjs';
 import {take} from 'rxjs/operators';
 
 export interface AlertmanagerConfigDialogData {
   title: string;
   projectId: string;
-  cluster: Cluster;
+  clusterId: Cluster;
```

Comment:
Is the `Cluster` object a `clusterId`? The second one directly indicates some kind of ID string, not an object.

Reply:
in the constructor we declare a data variable to be AlertmanagerConfigDialogData and it seems that don't have a cluster key in the object when i console it it shows that its clusterId, the edit alert manager config isn't working in dev.kubermatic.io/ it shows an error because this.data.cluster is undefined in data so it can't read id.
![Screenshot from 2022-04-22 05-46-10](https://user-images.githubusercontent.com/85109141/164586541-f2e8a724-b912-4b16-8467-8387d0e5f6ea.png)

Reply:
@Inject(MAT_DIALOG_DATA) public data: AlertmanagerConfigDialogData
this is a log for data :
![Screenshot from 2022-04-22 06-00-56](https://user-images.githubusercontent.com/85109141/164588379-037e1018-9568-454d-a590-e29baa350765.png)

Reply:
Why was the name changed at all since it is an object?

Reply:
im not sure that i get your point ?

Reply:
```
cluster: Cluster;
```

was changed to

```
clusterId: Cluster;
```

`clusterId` suggests a `string` type. It is an object of type `Cluster`. The new name is misleading and should be reverted.